.eslintrc

```diff
@@ -73,6 +73,8 @@
       "error",
       "always"
     ],
+    "no-restricted-globals": [
+      "error", "addEventListener", "blur", "close", "closed", "confirm", "defaultStatus", "event", "external", "defaultstatus", "find", "focus", "frameElement", "frames", "history", "innerHeight", "innerWidth", "length", "location", "locationbar", "menubar", "moveBy", "moveTo", "name", "onblur", "onerror", "onfocus", "onload", "onresize", "onunload", "open", "opener", "opera", "outerHeight", "outerWidth", "pageXOffset", "pageYOffset", "parent", "print", "removeEventListener", "resizeBy", "resizeTo", "screen", "screenLeft", "screenTop", "screenX", "screenY", "scroll", "scrollbars", "scrollBy", "scrollTo", "scrollX", "scrollY", "self", "sessionStorage", "status", "statusbar", "stop", "toolbar", "top"],
```

Comment:
Just to check, what is this list representing (and should it be so big? Wondering if we can remove anything.)

Reply:
The list represents the common global variables which should not be used directly. I found the list from [here](https://github.com/sidoshi/eslint-restricted-globals/blob/master/index.js).

I did not use the [package](https://github.com/sidoshi/eslint-restricted-globals) directly since it is not maintained. Also, in the usage rules for this rule, the prescribed way was to add a list in eslintrc with the names of the globals we want to avoid. We can remove the globals which we don't think we need in the codebase. I am not sure how to find out such globals and remove from the list. Can you guide me here, Thanks!

Reply:
Ah I see, so this is the list we want to avoid. In that case, that sounds good! I was worried that it was a list of what we were willing to accept, and was concerned that the list of exceptions seemed to be so big. No further change needed here, then.

Thanks @ankita240796 !

Reply:
You're welcome!

Reply:
Can we also add document, console etc. in this list as we expect to use $log, $document?

Reply:
I don't think so because that is not the case for [restricted globals](https://eslint.org/docs/rules/no-restricted-globals). The rule disallows using global variables directly. For instance, `location` should be prefixed by `$window` but there is no such case with console. Does that sound good?

Reply:
@ankita240796 What do you mean by "I did not use the package directly since it is not maintained." it is true that the last commit was made 7 months ago but that doesn't necessarily mean that the package is not maintained. Are we adding any values to that list anyway?

Reply:
Yes, we are adding new values like sessionStorage and we might need to do that in future as well. So, it is better to add the list manually here.

Reply:
Hi @ankita240796, can we concatenate both the lists, that is, [this one](https://github.com/sidoshi/eslint-restricted-globals) and the one we are manually constructing, as specified in the usage [here](https://github.com/sidoshi/eslint-restricted-globals#usage)?

Reply:
The concatenation was not working in eslintrc. Is there any specific way to do that?

Reply:
I see. To make it work, you'll have to change `.eslintrc` to `.eslintrc.js` and put everything within `module.exports`. You can refer to [this](https://gist.github.com/adrianhall/70d63d225e536b4563b2) and [this](https://eslint.org/docs/user-guide/configuring#using-eslintrecommended) for example. Thanks!

Reply:
Hi folks -- a question for y'all. What's wrong with copying the list and maintaining our copy of it? It seems nice to not have to rely on any dependencies.

Reply:
Yes, it seems fine to me. Just one concern -- we'll have to update our copy of the list after some interval probably, right?

Reply:
Yes, if we plan to add some new globals. I am not sure if the eslint-restricted-globals package is being maintained a lot since the last commit was 7 months ago.

Reply:
I think maintaining our own copy has its own pros and cons, but let's not dive into that. I'm okay with having our own copy, but maybe it would be nice to make an PR on the eslint-restricted-globals package (if the globals that we are adding are universal).

Reply:
I agree with you on making a PR for the package. I will do that with the globals I have added in this PR as well.